mlx/backend/metal/indexing.cpp

```diff
@@ -142,7 +142,45 @@ void Scatter::eval_gpu(const std::vector<array>& inputs, array& out) {
   // Get kernel name
   std::ostringstream kname;
   std::string idx_type_name = nidx ? type_to_name(inputs[1]) : "";
-  kname << "scatter" << type_to_name(out) << idx_type_name;
+
+  int idx_ndim = nidx ? inputs[1].ndim() : 0;
+  bool index_nd1_specialization = (idx_ndim == 1);
+
+  // Bail from fast path (1d index specialization) if scatter dims aren't
+  // the outermost dims and contiguous since update access won't be raster
+  // order.
+  for (auto i = 0; i < axes_.size() && index_nd1_specialization; i++) {
+    if (axes_[i] != i) {
+      index_nd1_specialization = false;
+      break;
+    }
+  }
```

Comment:
Makes sense. 

Just making a note, but I think (as a follow up) we should provide sorted `axes_` and `indices` to the primitive. Right now scatter let's you pass in out of order indices so if you sent `{1, 0}` that would fail this check even though it would be fine..  It would also work better for graph optimization with compile.